includes/class-wc-payments.php

```diff
@@ -1569,41 +1569,52 @@ public static function maybe_init_woopay_direct_checkout() {
 	}
 
 	/**
-	 * Enqueues the common config script if the express checkout button is disabled on the cart page.
+	 * Validates whether the common config script should be enqueued and enqueues it.
+	 *
+	 * If the express checkout button is disabled on the cart page, the common config
+	 * script needs to be enqueued to ensure `wcpayConfig` is available on the cart page.
 	 *
 	 * @return void
 	 */
-	public static function enqueue_woopay_common_config_script() {
+	public static function validate_and_enqueue_woopay_common_config_script() {
```

Comment:
I don't like the name of this function much, however I was a bit out of options since `maybe_enqueue_woopay_common_config_script` is taken.

For clarity, it's not possible to move this check to the `maybe_enqueue_woopay_common_config_script` function because it generates a warning saying `is_page` was called incorrectly. That's why I needed to keep this logic separate.